### DISCUSSION TITLE
Is perfutils-java still being supported?

### DISCUSSION THREAD

**Discussion by leomeza1** [OpenSatKit/OpenSatKit#109] [General]
Hello,

I'm trying to run CPM.jar on my Pop_OS 22.04 (it's basically Ubuntu 22.04), using this command:
java -jar CPM.jar

But, I'm getting an exception as shown in the attached text file.  I've installed jcommon and jfreechart. I can't figure out if I need to pass the classpath in as a parameter, or if my Java environment is just not supported, or something else. Any suggestions on how to resolve this exception?
[cpm_exception.txt](https://github.com/OpenSatKit/OpenSatKit/files/11194819/cpm_exception.txt)

So, I tried to compile the Java source code in the perfutils-java folder, but it looks like the source code may be out-of-date or something. For example, CPMCFSSearchDialog.java tries to import a bunch of constants from CPMConstants.java, but some of the constants are missing from the CPMConstants.java file. Is this repo the best place for me to get the latest perfutils-java source code? Is there a fork or .zip file with the complete source?

Thanks in advance,
Leo

**Comment by open-stemware** [comment]
Even though the CPM source code is contained within OSK, I've never had to generate the JAR file and consider OSK a recipient of a product.  My understanding is that NASA JSC is not currently maintaining an open-source release.  Here's a snapshot of the [CPM GitHub](https://github.com/nasa/perfutils-java) readme. Note the NASA cFS repo tools directory does not have a copy of the CPM.


![image](https://github.com/OpenSatKit/OpenSatKit/assets/90527234/0bd572f6-e43f-4eff-ab7d-f9c69e92cacc)
